### PR TITLE
DOC: Various doc fixes

### PR DESCRIPTION
**Pull request by TomAugspurger**
- Fixed spacing
- Fixed method reference
- Fixed list line wrapping
- Fixed unbalanced ticks
- Fixed section-heading without colon
- Changed Interval Properties -> Attributes
- Changed Styler properties

xref https://github.com/pandas-dev/pandas/issues/16033

Will probably do more as followups, so let's keep that open.

### CODE REVIEW

File: doc/source/whatsnew/v0.20.0.txt [hunk 1]
```diff
@@ -1357,9 +1357,9 @@ If indicated, a deprecation warning will be issued if you reference theses modul
     "pandas._testing", "pandas.util.libtesting", ""
     "pandas._window", "pandas.core.libwindow", ""
 
-- The function :func:`~pandas.api.type.union_categoricals` is now importable from ``pandas.api.types``, formerly from ``pandas.types.concat`` (:issue:`15998`)
+- The function :func:`~pandas.api.type.union_categoricals` is now importable from ``pandas.api.types``, formerly from ```pandas.types.concat`` (:issue:`15998`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.20.0.txt]: also pandas.api.type.union_categoricals -> pandas.api.type**s**.union_categoricals

File: pandas/core/generic.py [hunk 2]
```diff
@@ -4210,14 +4210,14 @@ def groupby(self, by=None, axis=0, level=None, as_index=True, sort=True,
...
-            aligned; see ``.align()`` method). If ndarray is passed, the
-            values as-is determine the groups.
+        by : mapping, function, str, or iterable
+            Used to determine the groups for the groupby.
+            If ``by`` is a function, it's called on each value of the object's
+            index. If a dict or Series is passed, the Series or dict VALUES
+            will be used to determine the groups (the Series' values are first
+            aligned; see ``.align()`` method). If an ndarray is passed, the
+            values as-is determine the groups. A str or list of strs maybe be
+            used used to group by the columns in ``self``.
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: used used

File: pandas/core/generic.py [hunk 3]
```diff
@@ -4210,14 +4210,14 @@ def groupby(self, by=None, axis=0, level=None, as_index=True, sort=True,
...
-            used to determine the groups (the Series' values are first
-            aligned; see ``.align()`` method). If ndarray is passed, the
-            values as-is determine the groups.
+        by : mapping, function, str, or iterable
+            Used to determine the groups for the groupby.
+            If ``by`` is a function, it's called on each value of the object's
+            index. If a dict or Series is passed, the Series or dict VALUES
+            will be used to determine the groups (the Series' values are first
+            aligned; see ``.align()`` method). If an ndarray is passed, the
+            values as-is determine the groups. A str or list of strs maybe be
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: maybe -> may be ?

**Comment by jorisvandenbossche** [comment]
@TomAugspurger maybe leave the changes to the html repr for a separate PR?

**Comment by TomAugspurger** [comment]
Sorry, when I pushed I didn't realize I was experimenting on this branch.
Be previous commit removed one of the changes, but I missed `core/frame.py`

**Comment by TomAugspurger** [comment]
Doc build passed. Remaining warnings

```
WARNING: could not import _license.show_info
[31mWARNING: while setting up extension conf.py: directive 'autosummary' is already registered, it will be overridden
[31mWARNING: [autosummary] failed to import 'pandas.DataFrame.columns': no module named pandas.DataFrame.columns
[39;49;00m[31mWARNING: [autosummary] failed to import 'pandas.DataFrame.index': no module named pandas.DataFrame.index
[39;49;00m[31mWARNING: [autosummary] failed to import 'pandas.Panel.items': no module named pandas.Panel.items
[39;49;00m[31mWARNING: [autosummary] failed to import 'pandas.Panel.major_axis': no module named pandas.Panel.major_axis
[39;49;00m[31mWARNING: [autosummary] failed to import 'pandas.Panel.minor_axis': no module named pandas.Panel.minor_axis
[39;49;00m[31mWARNING: [autosummary] failed to import 'pandas.Series.index': no module named pandas.Series.index
[31m/tmp/doc/source/generated/pandas.IntervalIndex.rst:57: WARNING: toctree references unknown document 'generated/pandas.IntervalIndex.str'
[39;49;00m[31m/home/travis/build/pandas-dev/pandas/pandas/core/groupby.py:docstring of pandas.core.groupby.DataFrameGroupBy.describe:16: WARNING: Block quote ends without a blank line; unexpected unindent.
[39;49;00m[31m/home/travis/build/pandas-dev/pandas/pandas/core/groupby.py:docstring of pandas.core.groupby.SeriesGroupBy.nlargest:12: WARNING: Block quote ends without a blank line; unexpected unindent.
[39;49;00m[31m/home/travis/build/pandas-dev/pandas/pandas/core/groupby.py:docstring of pandas.core.groupby.SeriesGroupBy.nsmallest:12: WARNING: Block quote ends without a blank line; unexpected unindent.
[39;49;00m[31m/home/travis/build/pandas-dev/pandas/pandas/io/parsers.py:docstring of pandas.read_csv:158: WARNING: Inline interpreted text or phrase reference start-string without end-string.
[39;49;00m[31m/home/travis/build/pandas-dev/pandas/pandas/io/parsers.py:docstring of pandas.read_fwf:157: WARNING: Inline interpreted text or phrase reference start-string without end-string.
[39;49;00m[31m/home/travis/build/pandas-dev/pandas/pandas/io/parsers.py:docstring of pandas.read_table:158: WARNING: Inline interpreted text or phrase reference start-string without end-string.
[39;49;00m[31m/tmp/doc/source/overview.rst:120: WARNING: Include file '/tmp/LICENSE' not found or reading it failed
[39;49;00m[31minternal padding after source/whatsnew/v0.8.0.txt:15: WARNING: image file not readable: ../_static/whatsnew_secondary_y.png
[39;49;00m[31minternal padding after source/whatsnew/v0.8.0.txt:19: WARNING: image file not readable: ../_static/whatsnew_kde.png
[31mNone:None: WARNING: toctree contains reference to nonexisting document 'generated/pandas.IntervalIndex.str'
[39;49;00m[31m/home/travis/build/pandas-dev/pandas/pandas/core/indexes/interval.py:docstring of pandas.IntervalIndex:None: WARNING: toctree contains reference to nonexisting document 'generated/pandas.IntervalIndex.str'
[39;49;00m[31mNone:None: WARNING: toctree contains reference to nonexisting document 'generated/pandas.IntervalIndex.str'
[39;49;00m[31msource/whatsnew/v0.19.0.txt:482: WARNING: undefined label: io.bigquery_authentication (if the link has no caption the label must precede a section header)
[39;49;00m[31msource/whatsnew/v0.17.1.txt:60: WARNING: undefined label: style (if the link has no caption the label must precede a section header)
```

I might have a fix for the DataFrame.index, DataFrame.columns warnings, but not before the release.

I'm not really sure about the IntervalIndex.str warnings.

**Comment by jorisvandenbossche** [comment]
@TomAugspurger OK to merge this one? (minus the remaining change to the html)

**Comment by TomAugspurger** [comment]
https://github.com/pandas-dev/pandas/pull/16035/commits/0b6796173d60dfde499c009da980585120ea75e0 removed that, but if you want to double check before merging.

**Comment by jorisvandenbossche** [comment]
thanks!

How did you fix the Styler warnings in the end? As I see you reverted what you changed regarding the properties

**Comment by TomAugspurger** [comment]
`template:` -> `template :` 😄 

The properties were nice for the docs, but you couldn't use them as class attributes anymore, which is convenient.

**Comment by jorisvandenbossche** [comment]
> template: -> template : :smile:

rst .. (or numpydoc maybe in this case) :-)

**Pull request merged**